DataFormats/FWLite/interface/DataGetterHelper.h

```diff
@@ -57,7 +58,7 @@ namespace fwlite {
                              std::shared_ptr<HistoryGetterBase> historyGetter,
                              std::shared_ptr<BranchMapReader> branchMap = std::shared_ptr<BranchMapReader>(),
                              std::shared_ptr<edm::EDProductGetter> getter = std::shared_ptr<edm::EDProductGetter>(),
-                             bool useCache = false);
+                             bool useCache = false, std::function<void (TBranch*)> baFoo = [](TBranch*){});
```

Comment:
change to `std::function<void (TBranch const&)>` since we do not want them modifying the branch and the branch can never be `nullptr`.